### PR TITLE
[cli] Setup Unistyles theme

### PR DESCRIPTION
**Pull request by coyksdev**
## Description

- Add components to the unistyles theme that can be reused by each template
- Add missing unistyles registry import

### CODE REVIEW

File: cli/src/templates/packages/expo-router/drawer/app/(drawer)/index.tsx.ejs [hunk 1]
```diff
@@ -5,17 +5,23 @@
...
 <% } else if (props.stylingPackage?.name === "stylesheet") { %>
   import { StyleSheet, Text, View } from "react-native";
 <% } %>
 
 const Page = () => {
   <% if (props.stylingPackage?.name === "unistyles") { %>
-      const {styles} = useStyles(stylesheet)
-  <% } %>
-  <% if (props.stylingPackage?.name === "nativewind") { %>
+    const {theme} = useStyles()
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/drawer/app/(drawer)/index.tsx.ejs:16]:
```suggestion
    const { theme } = useStyles()
```

File: cli/src/templates/packages/expo-router/drawer/app/(drawer)/news.tsx.ejs [hunk 2]
```diff
@@ -5,18 +5,23 @@
...
   import { StyleSheet, Text, View } from "react-native";
 <% } %>
 
 const Page = () => {
   <% if (props.stylingPackage?.name === "unistyles") { %>
-    const {styles} = useStyles(stylesheet)
-  <% } %>
-  
-  <% if (props.stylingPackage?.name === "nativewind") { %>
+    const {theme} = useStyles();
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/drawer/app/(drawer)/news.tsx.ejs:16]:
```suggestion
    const { theme } = useStyles();
```

File: cli/src/templates/packages/expo-router/stack/app/details.tsx.ejs [hunk 3]
```diff
@@ -22,10 +22,17 @@ export default function Details() {
   const router = useRouter();
 
   <% if (props.stylingPackage?.name === "unistyles") { %>
-    const {styles} = useStyles(stylesheet);
-  <% } %>
+    const {styles, theme} = useStyles(stylesheet);
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/details.tsx.ejs:25]:
```suggestion
    const { styles, theme } = useStyles(stylesheet);
```

File: cli/src/templates/packages/expo-router/stack/app/details.tsx.ejs [hunk 4]
```diff
@@ -22,10 +22,17 @@ export default function Details() {
...
 
-	<% if (props.stylingPackage?.name === "nativewind") { %>
+    const BackButton = () => (
+			<TouchableOpacity onPress={router.back}>
+			<View style={styles.backButton}>
+				<Feather name="chevron-left" size={16} color={theme.colors.azureRadiance} />
+				<Text style={styles.backButtonText}>Back</Text>
+			</View>
+			</TouchableOpacity>
+		);
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/details.tsx.ejs:34]:
```suggestion
    const BackButton = () => (
			<TouchableOpacity onPress={router.back}>
			  <View style={styles.backButton}>
				  <Feather name="chevron-left" size={16} color={theme.colors.azureRadiance} />
				  <Text style={styles.backButtonText}>Back</Text>
			  </View>
			</TouchableOpacity>
		);
```

File: cli/src/templates/packages/expo-router/stack/app/index.tsx.ejs [hunk 5]
```diff
@@ -22,7 +22,7 @@ export default function Page() {
   <% } %>
 
   <% if (props.stylingPackage?.name === "unistyles") { %>
-    const {styles} = useStyles(stylesheet);
+    const {styles, theme} = useStyles(stylesheet);
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/index.tsx.ejs:25]:
```suggestion
    const { styles, theme } = useStyles(stylesheet);
```

File: cli/src/templates/packages/expo-router/stack/app/index.tsx.ejs [hunk 6]
```diff
@@ -77,6 +77,21 @@ export default function Page() {
...
+					<Text style={theme.components.title}>Hello World</Text>
+					<Text style={theme.components.subtitle}>This is the first page of your app.</Text>
+				</View>
+				<Link href={{ pathname: "/details", params: { name: "Dan" } }} asChild>
+					<TouchableOpacity style={theme.components.button} >
+					<Text style={theme.components.buttonText}>Show Details</Text>
+					</TouchableOpacity>
+				</Link>
+				</View>
+			</View>
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/stack/app/index.tsx.ejs:94]:
```suggestion
    <% } else if (props.stylingPackage?.name === "unistyles") { %>
      <View style={styles.container}>
				<View style={styles.main}>
          <Stack.Screen options={{ title: "Overview" }} />
				  <View>
					  <Text style={theme.components.title}>Hello World</Text>
					  <Text style={theme.components.subtitle}>This is the first page of your app.</Text>
				  </View>
				  <Link href={{ pathname: "/details", params: { name: "Dan" } }} asChild>
					  <TouchableOpacity style={theme.components.button} >
					    <Text style={theme.components.buttonText}>Show Details</Text>
					  </TouchableOpacity>
				  </Link>
				</View>
			</View>
```

**Review by danstepanov — CHANGES_REQUESTED** [review]
Few syntax changes

**Review by ludwig-pro — APPROVED** [review]
Great job! It's "DRY"! +253 −330

Just a suggestion: sometimes you are using single quotes ' and other times double quotes ". I guess single quotes are the way to follow according to the prettier.rc config

**Comment by danstepanov** [comment]
> Great job! It's "DRY"! +253 −330
> 
> Just a suggestion: sometimes you are using single quotes ' and other times double quotes ". I guess single quotes are the way to follow according to the prettier.rc config

yea please change all uses of double quotes to single quotes

**Comment by coyksdev** [comment]
> > Great job! It's "DRY"! +253 −330
> > Just a suggestion: sometimes you are using single quotes ' and other times double quotes ". I guess single quotes are the way to follow according to the prettier.rc config
> 
> yea please change all uses of double quotes to single quotes

@danstepanov would you think I create a separate PR for this? I will apply it to all of the templates

**Comment by danstepanov** [comment]
> > > Great job! It's "DRY"! +253 −330
> > > Just a suggestion: sometimes you are using single quotes ' and other times double quotes ". I guess single quotes are the way to follow according to the prettier.rc config
> > 
> > 
> > yea please change all uses of double quotes to single quotes
> 
> @danstepanov would you think I create a separate PR for this? I will apply it to all of the templates

just apply it to the files in your PR and address the above changes that I highlighted, no need for another PR

**Comment by coyksdev** [comment]
@danstepanov @ludwig-pro requested changes applied

**Review by danstepanov — APPROVED** [review]
LGTM

**Pull request merged**